tests/sentry/deletions/test_project.py

```diff
@@ -212,3 +223,44 @@ def test_delete_with_uptime_monitors(self):
         assert not ProjectUptimeSubscription.objects.filter(
             id=project_uptime_subscription.id
         ).exists()
+
+    def test_delete_with_workflow_engine_models(self):
+        project = self.create_project(name="test")
+
+        data_condition_group = self.create_data_condition_group()
+        data_condition = self.create_data_condition(condition_group=data_condition_group)
+        snuba_query = self.create_snuba_query()
+        subscription = QuerySubscription.objects.create(
+            project=project,
+            status=QuerySubscription.Status.ACTIVE.value,
+            subscription_id="123",
+            snuba_query=snuba_query,
+        )
+        data_source = self.create_data_source(
+            organization=self.organization, source_id=subscription.id
+        )
+        detector = self.create_detector(
+            project_id=project.id,
+            name="Test Detector",
+            type=MetricIssue.slug,
+            workflow_condition_group=data_condition_group,
+        )
+        workflow = self.create_workflow()
+        data_source_detector = self.create_data_source_detector(
+            data_source=data_source, detector=detector
+        )
+        detector_workflow = DetectorWorkflow.objects.create(detector=detector, workflow=workflow)
+
+        self.ScheduledDeletion.schedule(instance=project, days=0)
+
+        with self.tasks():
+            run_scheduled_deletions()
+
+        assert not Detector.objects.filter(id=detector.id).exists()
+        assert not DataSourceDetector.objects.filter(id=data_source_detector.id).exists()
+        assert not DetectorWorkflow.objects.filter(id=detector_workflow.id).exists()
+        assert not DataConditionGroup.objects.filter(id=data_condition_group.id).exists()
+        assert not DataCondition.objects.filter(id=data_condition.id).exists()
+        assert not DataSource.objects.filter(id=data_source.id).exists()
+        assert not QuerySubscription.objects.filter(id=subscription.id).exists()
+        assert not SnubaQuery.objects.filter(id=snuba_query.id).exists()
```

Comment:
might want to split these into separate cases - that way if multiple fail we'll be able to see all the cases that fail in a single test run. the way it's currently setup if a DataConditionGroup wasn't removed correctly, we wouldn't know if the DataSource was also removed correctly or not too.